### PR TITLE
Code review and refactoring

### PR DESCRIPTION
**Pull request by bharathkalyans**
1. Worked on Drifty_CLI, CreateLogs, FileLoader and ProgressBarThread.
2. Created package for the utility functions which were extracted from Drifty_CLI file.
3. Moved the HardCoded strings to DriftyConstant file.
4. Created Singletons for the Scanner class and CreateLogs file.

### CODE REVIEW

**Comment by bharathkalyans** [comment]
Used SonarLint (IntelliJ Plugin) to remove 90% of the code smells and followed best coding practices.

**Comment by SaptarshiSarkar12** [comment]
@bharathkalyans Issues help to manage the project and track it's progress. Please create an issue for the changes that you have brought through this Pull Request (PR).
I shall review the code changes within 3 days.
Thank you!

File: src/validation/DriftyValidation.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+     * @param input        Input String to validate.
+     * @param printMessage The message to print to re-input the confirmation.
+     * @return true if the user enters Y [Yes] and false if not.
+     */
+    public static boolean yesNoValidation(String input, String printMessage) {
+
+        while (true) {
+            if (input.length() == 0) {
+                System.out.println("Please enter Y for yes and N for no!");
+                logger.log("ERROR", "Please enter Y for yes and N for no!");
```
Comment (SaptarshiSarkar12) [inline src/validation/DriftyValidation.java:27]: Add a constant in `DriftyConstants.java` file and make the below changes.
```suggestion
                logger.log(LOGGER_ERROR, ENTER_Y_OR_N);
```

File: src/FileDownloader.java [hunk 2]
```diff
@@ -182,7 +102,7 @@ private static void downloadFile() {
                     FileOutputStream fos = new FileOutputStream(dir + fileName);
                     ProgressBarThread progressBarThread = new ProgressBarThread(fos, totalSize, fileName);
                     progressBarThread.start();
-                    Drifty_CLI.logger.log("INFO", "Downloading " + fileName + " ...");
+                    Drifty_CLI.logger.log(LOGGER_INFO, DOWNLOADING + fileName + " ...");
```
Comment (SaptarshiSarkar12) [inline src/FileDownloader.java:105]: I think adding a constant with name `DOWNLOADING` is no better than just keeping it as a `Downloading `(string).
```suggestion
                    Drifty_CLI.logger.log(LOGGER_INFO, "Downloading " + fileName + " ...");
```

File: src/Drifty_CLI.java [hunk 3]
```diff
@@ -1,291 +1,192 @@
...
-        System.out.println("Trying to auto-detect default Downloads folder...");
-        logger.log("INFO", "Trying to auto-detect default Downloads folder...");
-        if (!System.getProperty("os.name").contains("Windows")) {
-            String home = System.getProperty("user.home");
-            downloadsFolder = home + "/Downloads/";
-        }
-        else {
+    private static void saveToDefault() {
+        System.out.println(TRYING_TO_AUTOMATICALLY_DETECT_FILE);
+        logger.log(LOGGER_ERROR, TRYING_TO_AUTOMATICALLY_DETECT_FILE);
```
Comment (SaptarshiSarkar12) [inline src/Drifty_CLI.java:149]:
```suggestion
        logger.log(LOGGER_INFO, TRYING_TO_AUTO_DETECT_FILE);
```

File: src/Drifty_CLI.java [hunk 4]
```diff
@@ -1,291 +1,192 @@
...
-    private static void saveToDefault(){
-        System.out.println("Trying to auto-detect default Downloads folder...");
-        logger.log("INFO", "Trying to auto-detect default Downloads folder...");
-        if (!System.getProperty("os.name").contains("Windows")) {
-            String home = System.getProperty("user.home");
-            downloadsFolder = home + "/Downloads/";
-        }
-        else {
+    private static void saveToDefault() {
+        System.out.println(TRYING_TO_AUTOMATICALLY_DETECT_FILE);
```
Comment (SaptarshiSarkar12) [inline src/Drifty_CLI.java:148]: Please change `TRYING_TO_AUTOMATICALLY_DETECT_FILE` to `TRYING_TO_AUTO_DETECT_FILE` to reduce the name of the constant.
```suggestion
        System.out.println(TRYING_TO_AUTO_DETECT_FILE);
```

**Review by SaptarshiSarkar12 — COMMENTED** [review]
Please make the changes.

**Comment by SaptarshiSarkar12** [comment]
@bharathkalyans Let me know when you are done.

**Pull request closed without merging**